oeps/oep-0045/decisions/0001-tutor-as-replacement-for-edx-configuration.rst

```diff
@@ -0,0 +1,328 @@
+improvement. The company behind it, `Overhang.IO <https://overhang.io/>`__, is declaredly committed to open source
+values and offers `Tutor deployment <https://overhang.io/tutor/>`__ as a service, and thus have a financial incentive to
+continue maintaining it.
+
+It is, however, still effectively maintained `by a single developer
+<https://github.com/overhangio/tutor/graphs/contributors>`__: Regis Behmo. This poses a significant risk: if for some
+unforeseen reason Regis can no longer maintain Tutor, the Open edX community would again be orphaned. Thus, it is
+suggested that some form of co-governance/co-maintenance be instituted. Regis has `tentatively agreed
+<https://discuss.openedx.org/t/lets-talk-about-the-native-installation/3269/5>`__.  Hopefully, with more people and
+companies using Tutor, more contributors will be willing to take on this responsibility.
+
+Tutor's codebase is technically up-to-date, well organized, and maintained. There is no detectable technical debt.
+
+
+Conclusion of the evaluation
+............................
+
+
+Positives
+`````````
+
+At the time of writing, Tutor is the only existing alternative to ``edx/configuration``. Luckily for the community, it
+ticks a lot of the boxes that qualify a good replacement:
+
+-  It can do single-server production deployments well
+-  It allows for automated deployment as well as, or better than, ``edx/configuration``
+-  It supports Koa, and new releases are added quickly
+-  It allows for theme customization, running edx-platform forks, and installing custom requirements
+-  It is extensible via plugins
+-  Actively maintained by a committed developer; codebase has no technical debt
+-  Licensed under the AGPL v3
+
+
+Omissions
+`````````
+
+There is work to be done before it can be considered a full replacement:
+
+-  There must be a documented way to run highly-available, multi-server production deployments, including a way to run
+   automated rolling upgrades with no downtime
+-  Migration from the native installation must be documented and/or automated
+-  It must support deployments of the master branches of edx-platform and IDAs, and once that comes about it must
+   undergo CI/CD so that breakages can be dealt with quickly
+-  There should be more than one maintainer
+
+It is estimated that initial versions of the above will take anywhere from three to six months to achieve, if sufficient
+developers commit resources to the endeavor. (Part of the test of whether a community-supported deployment method will
+work hinges precisely on whether this community can pool resources to maintain it!)
+
+
+Wishlist
+````````
+
+The following would be great to have once the TODO list above is taken care of, as they're not currently present in
+either Tutor or ``edx/configuration``:
+
+-  Support the deployment of MFEs
+-  There should be a way to reuse the same base LMS image with different themes without having to bake them into the
```

Comment:
At the risk of asking an underinformed, handwavy question: couldn't this be achieved by using [Django `STATICFILES_STORAGE`](https://docs.djangoproject.com/en/3.1/howto/static-files/deployment/#staticfiles-from-cdn) on S3 or another external storage service? In other words, isn't this something that could be solved in edx-platform?

Reply:
Good idea. You should already be able to run `tutor local run lms paver update_assets`, with appropriate settings, and have the static assets sent to S3/MinIO. You'll need a tutor plugin to define the proper Django settings, though.

Reply:
You mean something like [https://github.com/hastexo/tutor-s3](this)? :) (I think the only thing that that's missing is setting `STATICFILES_STORAGE` itself.)

Reply:
Sorry, one other thing that that plugin is missing is updates for boto3.